batch/batch/server/server.py

```diff
@@ -130,10 +176,20 @@ def __init__(self, pod_spec, batch_id, attributes, callback, parent_ids, scratch
         self.parent_ids = parent_ids
         self.incomplete_parent_ids = set(self.parent_ids)
         self.scratch_folder = scratch_folder
+
         self._pod_name = None
         self.exit_code = None
         self._state = 'Created'
 
+        self._tasks = [JobTask.copy_task(self.id, 'input', input_files),
+                       JobTask(self.id, 'main', pod_spec),
+                       JobTask.copy_task(self.id, 'output', output_files)]
+
+        self._tasks = [t for t in self._tasks if t is not None]
+        self._task_idx = -1
+        self._current_task = self._next_task()
```

Comment:
let's not assign it since _next_task already does that. I prefer the consistency of always seeing `_next_task` as a mutative operation.